cpp/src/community/louvain.cu

```diff
@@ -26,50 +26,28 @@ namespace cugraph {
 namespace detail {
 
 template <typename vertex_t, typename edge_t, typename weight_t>
-std::pair<size_t, weight_t> louvain(raft::handle_t const &handle,
-                                    GraphCSRView<vertex_t, edge_t, weight_t> const &graph_view,
-                                    vertex_t *clustering,
-                                    size_t max_level,
-                                    weight_t resolution)
+std::pair<std::unique_ptr<Dendrogram<vertex_t>>, weight_t> louvain(
+  raft::handle_t const &handle,
+  GraphCSRView<vertex_t, edge_t, weight_t> const &graph_view,
```

Comment:
Yes.  The legacy version is still faster on SG than the new version.

Reply:
So, are we using the legacy graph for this? or are we using the legacy version for SG and the new version for MG?

Reply:
The C++ supports:
* Legacy graph for SG
* new graph for SG
* new graph for MG

The python is currently still using legacy graph for SG and new graph for MG.  I'm doing some performance analysis now of the new graph for SG vice the legacy graph for SG.  It might be necessary to get some optimizations in place in order for the new graph to be fast enough.  My last run today (haven't pushed the latest mods yet, but will shortly) had `web-Google.mtx` running in 14s on the legacy implementation and 56s on the new implementation.  `ljournal-2008.mtx` runs in 250s on legacy and crashes (OOM) on the new implementation.

Reply:
Oops, ljournal is not large at all and if you see OOM with ljournal, I guess something really shouldn't happen is happening in this case; something we should take a look (maybe not in this PR).

Reply:
I may dig deeper to both performance & memory footprint when I tune primitives for Louvain. The execution time and memory consumption sound very bad.

Reply:
We should be able to process 1B-ish edge graph using one GPU. IIRC, ljournal is much smaller than that.

Reply:
Agreed, that should probably be outside of this PR.  But that's why I've left the legacy implementation in place and the default for python.  When you start tuning these we can evaluate when to delete the legacy implementation.